src/openrct2-ui/drawing/engines/SoftwareDrawingEngine.cpp

```diff
@@ -121,32 +121,20 @@ class SoftwareDrawingEngine final : public X8DrawingEngine
             SDL_UnlockSurface(_surface);
         }
 
-        // Copy the surface to the window
-        if (gConfigGeneral.WindowScale == 1 || gConfigGeneral.WindowScale <= 0)
```

Comment:
Why did you remove this branch? Now there is always an extra copy.

Reply:
This was causing issues on highdpi on macos. See https://discord.com/channels/264137540670324737/691736850980667514/1227237592743940146. I am aware there is one more copy, but without access to the hardware in question I cannot fix it another way

Reply:
I can guard it for macos only

Reply:
Filed https://github.com/OpenRCT2/OpenRCT2/issues/21772 for addressing this properly - but I cannot do this myself due to lack of hardware required to reproduce it

Reply:
I see, fair enough.